src/main/java/com/example/demo/service/BookServiceImpl.java

```diff
@@ -1,4 +1,46 @@
 package com.example.demo.service;
 
-public class BookServiceImpl {
+import com.example.demo.dto.book.BookDto;
+import com.example.demo.dto.book.CreateBookRequestDto;
+import com.example.demo.mapper.BookMapper;
+import com.example.demo.model.Book;
+import com.example.demo.repository.BookRepository;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto bookDto) {
+        Book book = bookMapper.toModel(bookDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Can't find book by id " + id));
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto bookDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find book by id" + id));
+        bookMapper.updateModelFromDto(book, bookDto);
+        return bookMapper.toDto(bookRepository.save(book));
```

Comment:
do you use this method? your task also implement 
PUT: /api/books/{id} (Update a specific book)
DELETE /api/books/{id} (Delete a specific book) 
endpoints